ee/clickhouse/queries/trends/total_volume.py

```diff
@@ -38,48 +33,156 @@ def _total_volume_query(self, entity: Entity, filter: Filter, team_id: int) -> T
-                    {"aggregated_value": result[0][0] if result and len(result) else 0, "days": time_range}
-                ],
+        if entity.math in [WEEKLY_ACTIVE, MONTHLY_ACTIVE]:
+            final_query = ACTIVE_USER_SQL.format(
+                event_query=event_query,
+                **content_sql_params,
+                parsed_date_to=trend_event_query.parsed_date_to,
+                parsed_date_from=trend_event_query.parsed_date_from,
+                **trend_event_query.active_user_params
             )
         else:
+            final_query = VOLUME_SQL.format(event_query=event_query, **content_sql_params)
 
-            if entity.math in [WEEKLY_ACTIVE, MONTHLY_ACTIVE]:
-                content_sql = ACTIVE_USER_SQL.format(
-                    event_query=event_query,
-                    **content_sql_params,
-                    parsed_date_to=trend_event_query.parsed_date_to,
-                    parsed_date_from=trend_event_query.parsed_date_from,
-                    **trend_event_query.active_user_params
-                )
-            else:
-                content_sql = VOLUME_SQL.format(event_query=event_query, **content_sql_params)
-
-            null_sql = NULL_SQL.format(trunc_func=trunc_func, interval_func=interval_func)
-            params["date_from"] = format_ch_timestamp(filter.date_from or get_earliest_timestamp(team_id), filter)
-            params["date_to"] = format_ch_timestamp(filter.date_to, filter)
-            params["interval"] = filter.interval
-            final_query = AGGREGATE_SQL.format(null_sql=null_sql, content_sql=content_sql)
-            return final_query, params, self._parse_total_volume_result(filter)
+        params["date_from"] = format_ch_timestamp(filter.date_from or get_earliest_timestamp(team_id), filter)
+        params["date_to"] = format_ch_timestamp(filter.date_to, filter)
+        params["interval"] = filter.interval
+        return final_query, params, self._parse_total_volume_result(filter)
 
     def _parse_total_volume_result(self, filter: Filter) -> Callable:
         def _parse(result: List) -> List:
-            parsed_results = []
-            for _, stats in enumerate(result):
-                parsed_result = parse_response(stats, filter)
-                parsed_results.append(parsed_result)
+            # NOTE: there's some implicit coupling here:
+            #
+            #        1. the ordering of the fields we get back, (date, value)
+            #        2. that the last row is always the `WITH TOTALS` result
+            #
+            #  TODO: remove coupling
+
+            #  The last row relates to the `WITH TOTALS` from the clickhouse query
+            values_without_totals, (_, aggregated_value) = result[:-1], result[-1]
+
+            # Separate dates and values, as desired for the REST API response
+            dates, values = zip(*sorted(values_without_totals))
+
+            parsed_result = parse_response([dates, values], filter)
```

Comment:
The above 3 lines are much much harder to grok than what was going on previously. Why did this become so complicated? Can we simplify somehow?

Reply:
Right, there are a couple of changes that required updating here

 * I added `WITH TOTALS`
 * this with totals will be the last entry in the results
 * so I changed the query from returning results in the form `[[date1, date2, ...], [value1, value2, ...]]` to `[[date1, value1], [date2, value2], ...]`
 * parse_response is expecting the former format

Some options I could try:

 1. I can check to see if I can get `WITH TOTALS` to work with returning arrays as well
 2. I can change just remove the call to `parse_response` and do something simpler here instead

I'll go with the second unless there's any reason not to